ui/components/UserPreferences.js

```diff
@@ -42,59 +41,123 @@ const styles = (theme) => ({
-      backgroundColor : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
+      backgroundColor : theme.palette.type === "dark" ? "#00B39F"  : theme.palette.primary,
     },
   },
   tab : {
-    maxWidth : 'min(33%, 200px)',
-    minWidth : '50px',
+    maxWidth : "min(33%, 200px)",
+    minWidth : "50px",
     margin : 0,
     "&.Mui-selected" : {
-      color : theme.palette.type === 'dark' ? "#00B39F" : theme.palette.primary,
-    }
+      color : theme.palette.type === "dark" ? "#00B39F"  : theme.palette.primary,
+    },
   },
   icon : {
-    display : 'inline',
-    verticalAlign : 'text-top',
+    display : "inline",
+    verticalAlign : "text-top",
     width : theme.spacing(1.75),
     marginLeft : theme.spacing(0.5),
   },
   iconText : {
-    display : 'inline',
-    verticalAlign : 'middle',
+    display : "inline",
+    verticalAlign : "middle",
   },
-  backToPlay : { margin : theme.spacing(2), },
-  link : { cursor : 'pointer', },
+  backToPlay : { margin : theme.spacing(2) },
+  link : { cursor : "pointer" },
   formContainer : {
-    display : 'flex',
-    'flex-wrap' : 'wrap',
-    'justify-content' : 'space-evenly',
-    padding : 50
+    display : "flex",
+    "flex-wrap" : "wrap",
+    "justify-content" : "space-evenly",
+    padding : 50,
   },
   formGrp : {
     padding : 20,
-    border : '1.5px solid #969696',
+    border : "1.5px solid #969696",
+    margin : 20,
+    display : "flex",
+    "flex-direction" : "column",
+    width : "100%",
   },
-  formLegend : { fontSize : 20, },
+  formLegend : { fontSize : 20 },
+  formLegendSmall : { fontSize : 16 },
   switchBase : {
-    color : '#647881',
-    "&$checked" : { color : '#00b39f' },
-    "&$checked + $track" : { backgroundColor : 'rgba(0,179,159,0.5)' },
+    color : "#647881",
+    "&$checked" : { color : "#00b39f" },
+    "&$checked + $track" : { backgroundColor : "rgba(0,179,159,0.5)" },
   },
-  track : { backgroundColor : 'rgba(100,120,129,0.5)', },
+  track : { backgroundColor : "rgba(100,120,129,0.5)" },
   checked : {},
   tabLabel : {
     [theme.breakpoints.up("sm")] : {
-      fontSize : '1em'
+      fontSize : "1em",
     },
-    [theme.breakpoints.between("xs", 'sm')] : {
-      fontSize : '0.8em'
-    }
-  }
+    [theme.breakpoints.between("xs", "sm")] : {
+      fontSize : "0.8em",
+    },
+  },
 });
 
+function ThemeToggler({ theme, themeSetter, enqueueSnackbar, classes }) {
+  const [themeToggle, setthemeToggle] = useState(false);
+  const defaultTheme = "light";
+  const handle = () => {
+    if (isExtensionOpen()) {
+      return;
+    }
+
+    theme === "dark" ? setthemeToggle(true)  : setthemeToggle(false);
+    localStorage.setItem("Theme", theme);
+  };
+
+  useLayoutEffect(() => {
+    if (isExtensionOpen()) {
+      if (localStorage.getItem("Theme") && localStorage.getItem("Theme") !== defaultTheme) {
+        themeSetter(defaultTheme);
+      }
+      return;
+    }
+
+    if (localStorage.getItem("Theme") === null) {
```

Comment:
It would be nice if `Theme` is defined somewhere (in constant.js) and used globally.

Reply:
okay @MUzairS15 , so you want me to create a file for theme in the constants folder right

Reply:
Rather, define Theme in constants.js, so that we don’t hardcode “Theme” everywhere, but instead it can be referenced via the constant.

Reply:
okay sure will do that